palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/SimplePalisadeService.java

```diff
@@ -0,0 +1,257 @@
+                    return policyRequest;
+                })
+                .thenApply(req -> {
+                    LOGGER.debug("Getting policy from policyService: {}", req);
+                    return policyService.getPolicy(req).toCompletableFuture().join();
+                }).thenApply(policy -> {
+                    LOGGER.debug("Got policy: {}", policy);
+                    return policy;
+                });
+    }
+
+    private void auditRegisterRequestComplete(final RegisterDataRequest request, final User user, final MultiPolicy multiPolicy) {
+        RegisterRequestCompleteAuditRequest registerRequestCompleteAuditRequest = RegisterRequestCompleteAuditRequest.create(request.getOriginalRequestId())
+                .withUser(user)
+                .withLeafResources(multiPolicy.getPolicies().keySet())
+                .withContext(request.getContext());
+        LOGGER.debug("Auditing: {}", registerRequestCompleteAuditRequest);
+        auditService.audit(registerRequestCompleteAuditRequest).toCompletableFuture().join();
+    }
+
+    private void auditRequestReceivedException(final RegisterDataRequest request, final Throwable ex, final Class<? extends Service> serviceClass) {
+        final RegisterRequestExceptionAuditRequest auditRequestWithException =
+                RegisterRequestExceptionAuditRequest.create(request.getOriginalRequestId())
+                        .withUserId(request.getUserId())
+                        .withResourceId(request.getResourceId())
+                        .withContext(request.getContext())
+                        .withException(ex)
+                        .withServiceClass(serviceClass);
+        LOGGER.debug("Error handling: " + ex.getMessage());
+        auditService.audit(auditRequestWithException).toCompletableFuture().join();
+    }
+
+    private void cache(final RegisterDataRequest request, final User user,
+                       final RequestId requestId, final MultiPolicy multiPolicy,
+                       final int resCount,
+                       final RequestId originalRequestId) {
+        DataRequestConfig dataRequestConfig = new DataRequestConfig()
+                .user(user)
+                .context(request.getContext())
+                .rules(multiPolicy.getRuleMap());
+        dataRequestConfig.setOriginalRequestId(originalRequestId);
+        final AddCacheRequest<DataRequestConfig> cacheRequest = new AddCacheRequest<>()
+                .key(requestId.getId())
+                .value(dataRequestConfig)
+                .service(this.getClass());
+        LOGGER.debug("Caching: {}", cacheRequest);
+        final Boolean success = cacheService.add(cacheRequest).join();
+        if (null == success || !success) {
+            throw new CompletionException(new RuntimeException("Failed to cache request: " + request));
+        }
+        //set a quick count of how many resources are needed for this request
```

Comment:
@m09526 Is this section still required if we are removing the redirection service?

Reply:
No, this was used to provide a metric of how many resources are currently "in flight". Would be used to help autoscaling. I don't think this is the right way to do that anymore, so lets get rid of it.